### PR TITLE
Implement a new file logger API

### PR DESCRIPTION
**Pull request by pasin**
- Implemented a new file logger API a refactor log file related settings into a single configuration file, LogFileConfiguration.

- Updated unit tests.

#2321

### CODE REVIEW

File: Objective-C/CBLFileLogger.h [hunk 1]
```diff
@@ -19,41 +19,28 @@
...
-/**
- The directory to store the log files.
- */
-@property (nonatomic) NSString* directory;
-
-/**
- To use plain text file format instead of the default binary format.
+ The log file configuration for configuring the log directory, file format, and rotation policy.
+ The config property is nil by default. Setting the config property to nil will disable the file
+ logging.
```
[the remark this answers is not in the document]
Comment (pasin) [reply Objective-C/CBLFileLogger.h]: OK, PR is updated.

File: Objective-C/CBLLogFileConfiguration.m [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+- (void) checkReadonly {
+    if (_readonly) {
+        [NSException raise: NSInternalInconsistencyException
+                    format: @"This configuration object is readonly."];
+    }
+}
+
+
+- (NSString*) defaultDirectory {
```
Comment (jayahariv) [inline Objective-C/CBLLogFileConfiguration.m]: not seeing anywhere this is used, as well as an option to set directory for `CBLLogFileConfiguration`, intentional?

Comment (pasin) [reply Objective-C/CBLLogFileConfiguration.m]: I left it there before talking to @rajagp yesterday to confirm that we will not have default directory anymore. So the conclusion is that we will not have default directory anymore. We could remove that.

Comment (pasin) [reply Objective-C/CBLLogFileConfiguration.m]: PR updated.

File: Swift/LogFileConfiguration.swift [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+        self.readonly = readonly
+        self.directory = config.directory
+        self.usePlainText = config.usePlainText
+        self.maxSize = config.maxSize
+        self.maxRotateCount = config.maxRotateCount
+    }
+    
+    func checkReadOnly() {
+        if self.readonly {
+            fatalError("This configuration object is readonly.")
```
Comment (jayahariv) [inline Swift/LogFileConfiguration.swift]: did you mean 

```
if !readOnly {
  // show error
}
```

Comment (pasin) [reply Swift/LogFileConfiguration.swift]: No, only show if it's readonly. I think it's already correct.

File: Objective-C/CBLLogFileConfiguration.m [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+
+
+#pragma mark - Internal
+
+
+- (void) checkReadonly {
+    if (_readonly) {
+        [NSException raise: NSInternalInconsistencyException
+                    format: @"This configuration object is readonly."];
+    }
```
Comment (jayahariv) [inline Objective-C/CBLLogFileConfiguration.m]: understood what this mean by, but the method confused. may be adding a single line of comment saying, "will raise an exception when called on _readOnly". Changing the method name might crush the consistency else we should have renamed the method more relevant.

Comment (pasin) [reply Objective-C/CBLLogFileConfiguration.m]: I think method name is good enough. Another option might be `validateReadOnly`. I will add a comment to the method. Thanks!

**Pull request merged**